Fix conditional __AFL_LOOP function definition

unseddd opened the pull request:
`__AFL_LOOP` function name conflicts with macro definition in `AFL`/`AFLPlusPlus`.

Resolves #7230

moneromooo-monero commented:
That'd break cases where it's not defined. It'd be better to fix the #if to catch the correct cases.

Commit:
Fix conditional __AFL_LOOP definition

__AFL_LOOP function definition conflicts with macro definition.

Conditionally add __AFL_LOOP function definition only when macro
is not defined.

Resolves #7230

unseddd commented:
> That'd break cases where it's not defined. It'd be better to fix the #if to catch the correct cases.

Ah, yeah, I understand your point. Fixed to directly test whether `__AFL_LOOP` is defined.

moneromooo-monero commented:
Assuming it's always a define, looks good.

selsta commented:
This solution is a bit fragile, I have opened https://github.com/monero-project/monero/pull/10822 instead.
